battleEngine.cpp

```diff
@@ -1,7 +1,10 @@
 #include <iostream>
 #include "battleEngine.h"
+#include "rng.h"
 
-void BattleEngine::StateMachine::commenceBattle(Monster* a, Monster* b)
+Random randomizing;
```

Comment:
Global instantiation, not good. Same as other suggestion, move inside of the battle engine class